floris/wind_data.py

```diff
@@ -405,6 +427,110 @@ def resample(self, wd_step=None, ws_step=None, inplace=False):
         else:
             return resampled_wind_rose
 
+    def resample_by_interpolation(self, wd_step=None, ws_step=None, method="linear", inplace=False):
+        """
+
+        Resample the wind rose using interpolation.  The method can be either
+        'linear' or 'nearest'.  If inplace is set to True, the current WindRose
+        object will be updated with the resampled bins.
+
+        Args:
+            wd_step: Step size for wind direction resampling (float, optional).
+            ws_step: Step size for wind speed resampling (float, optional).
+            method: Interpolation method to use.  Can be either 'linear' or 'nearest'.
+            inplace: Flag indicating whether to update the current WindRose object.
+
+        Returns:
+            WindRose: Resampled wind rose based on the provided or default step sizes.
+
+        """
+        if method == "linear":
+            interpolator = LinearNDInterpolator
+        elif method == "nearest":
+            interpolator = NearestNDInterpolator
+        else:
+            UserWarning("Unknown interpolation method: '{:s}'".format(method))
```

Comment:
```python
UserWarning("Unknown interpolation method: '{:s}'. Available methods are 'linear' and 'nearest'".format(method))
```

Reply:
fixed as well

Reply:
Actually what is a "UserWarning"? does it cause an error? If not, what happens next?

Reply:
Oof, this should be a ValueError, switching